pkg/cri/resource-manager/requests.go

```diff
@@ -60,6 +60,19 @@ func (m *resmgr) startRequestProcessing() error {
 		return err
 	}
 
+	//
+	// Notes:
+	//   While normally it is enough to release stale containers and allocate
+	//   newly discovered ones, if we are switching policies we need to force
+	//   reallocating everything. Otherwise containers already present in the
+	//   cache would not get properly updated by the new policy.
+	//
+	if m.policySwitch {
+		containers := m.cache.GetContainers()
+		cache.SortContainers(containers)
+		add, del = containers, containers
+	}
```

Comment:
Would the if branch be good place to add
```
m.policySwitch = false
```
...some other place might do as well. But I'm somehow missing that line...

Reply:
> One question I started to wonder (I don't remember the code on top of my head) is that do we add the new containers in random or some meaningful order? If it's not already that way, would it be easy sort the containers by Pod id (could be left as a future improvement)?

Since we disrupt containers here we might as well try to optimize their placement. Therefore this patch
strongly suggests for the active policy to reallocate containers in most-important-largest-first order by
sorting the container slice by QoS/memory/CPU before passing it on to the policy.

Reply:
> Would the if branch be good place to add
> 
> ```
> m.policySwitch = false
> ```
> 
> ...some other place might do as well. But I'm somehow missing that line...

Thanks for reminding me. it was supposed to be there for clarity and regularity, regardless
of this code never being run twice, but it slipped my mind.

Reply:
> Since we disrupt containers here we might as well try to optimize their placement. Therefore this patch
> strongly suggests for the active policy to reallocate containers in most-important-largest-first order by
> sorting the container slice by QoS/memory/CPU before passing it on to the policy.

Ah, sorry, I somehow missed the `SortContainers()` in this patch 😬 All good 😄